Update constituencies for the 2021 Bundestag Election

azuser opened the pull request:
Reference: https://www.bundeswahlleiter.de/en/bundestagswahlen/2021/wahlkreiseinteilung.html#main

azuser commented:
@jloutsenhizer @sguenther85 can you please have a look ?

jloutsenhizer commented on identifiers/country-de/aliases.csv:
```diff
@@ -214,7 +214,7 @@ ocd-division/country:de/land:nw/wahlkreis:heinsberg,ocd-division/country:de/stat
 ocd-division/country:de/land:nw/wahlkreis:herne_~_bochum_ii,ocd-division/country:de/state:nw/cd:141,Added for backwards compatibility with land/wahlkreis identifiers
 ocd-division/country:de/land:nw/wahlkreis:hochsauerlandkreis,ocd-division/country:de/state:nw/cd:147,Added for backwards compatibility with land/wahlkreis identifiers
-ocd-division/country:de/land:nw/wahlkreis:höxter_~_lippe_ii,ocd-division/country:de/state:nw/cd:136,Added for backwards compatibility with land/wahlkreis identifiers
+ocd-division/country:de/land:nw/wahlkreis:höxter_~_gütersloh_iii_~_lippe_ii,ocd-division/country:de/state:nw/cd:136,Added for backwards compatibility with land/wahlkreis identifiers
```
I think these should be added as additional aliases rather than replacing the old ones

azuser replied:
If we do so, the old identifier ocd-division/country:de/land:nw/wahlkreis:höxter_~_lippe_ii would be considered as correct.

jloutsenhizer replied:
perhaps we could add validTo date to the old aliased identifiers then?

My concern is that if some data has the OCD ID `ocd-division/country:de/land:nw/wahlkreis:höxter_~_lippe_ii` attached to it, then there's no way to know that it's actually the same as `ocd-division/country:de/state:nw/cd:136` if we're modifying the line.

azuser replied:
Good idea. I updated the aliases.

jloutsenhizer reviewed (commented):
Not sure why github is showing the wahlkreise.csv file as being completely changed, the two lines changed lgtm. One comment around the aliases and then I think this change will be good

sguenther85 commented:
hi, was the complete last week on "vacation" (or how you call it in the current time). ;) 

i'll look over it again and get back to you. i also have another question according to ocds and the Bundestagswahl.

sguenther85 commented:
Hi, both changes looking good for me.
But after we have discussed and adjusted the assembly and the meaning of an ocd-path several times in the last 4 years, the "state" level here would actually be wrong, because they are geographical.

So for the "german feder election" itself it must be onyl the ed's.
But i'm very interested in list the constituencies of the "german fedaral election" and alos of an "Saxony-Anhalt state election" (Landtagswahl in Sachsen-Anhalt) for exmaple. Because we have this state election in June.

It is possible to list for several different level elections the ocd-paths in one file? Its a little bit the same problem we have for UK (GB) where we tried to list constituencies for UK Election and Scotland Election.

In this case we have here also the same problem that we have two times the same constituency "name" for two election, but with an different cut on the map.

azuser commented:
> Hi, both changes looking good for me.
> But after we have discussed and adjusted the assembly and the meaning of an ocd-path several times in the last 4 years, the "state" level here would actually be wrong, because they are geographical.
> 
> So for the "german feder election" itself it must be onyl the ed's.
Can you please provide more details ? 

As explained in the [wikipedia page](https://en.wikipedia.org/wiki/2021_German_federal_election), If a party wins fewer constituency seats in a state than its second votes would entitle it to, it receives additional seats from the relevant state list. 

So, I guess, the states level is important to represent the electoral system in that case.

> But i'm very interested in list the constituencies of the "german fedaral election" and alos of an "Saxony-Anhalt state election" (Landtagswahl in Sachsen-Anhalt) for exmaple. Because we have this state election in June.
> 
> It is possible to list for several different level elections the ocd-paths in one file? Its a little bit the same problem we have for UK (GB) where we tried to list constituencies for UK Election and Scotland Election.
> 
> In this case we have here also the same problem that we have two times the same constituency "name" for two election, but with an different cut on the map.

Hello, 

Can you please

jdmgoogle commented:
@azuser In previous threads the discussion has focused on not how seats are allocated within the electoral system, but which governmental body has authority over creating the geographic entities [[1](https://github.com/opencivicdata/ocd-division-ids/pull/188#issuecomment-602206693), [2](https://github.com/opencivicdata/ocd-division-ids/issues/184#issuecomment-555026889), [3](https://github.com/opencivicdata/ocd-division-ids/pull/181#issuecomment-549548955)]

So the principle here is not how any particular electoral system uses the boundaries or geographic entities, but at which level of government they are created and updated.

@jpmckinney for another opinion, though.

jpmckinney commented:
In Canada, the constitution allocates seats in the federal legislature (each of which has a corresponding division) to each province (single-seat ridings). A boundary commission then sets the boundaries of the divisions. Canada has provincial legislatures, which are not required to use the same number of seats or shape of divisions as the federal level, though some do. The provincial electoral divisions are set by different legislation and boundary commissions.

As such, in Canada, federal divisions are at ocd-division/country:ca/ed: whereas provincial divisions are at ocd-division/country:ca/province:on/ed:, etc.

It seems like the situation is similar in Germany, so a similar model should be used.

Regarding the use of party lists for state divisions, you can just use the state division ocd-division/country:de/land:xx if needed. There is no need to nest federal divisions under `land`. The OCDID is not intended to encode that kind of relationship. If, for example, Germany switched back to only first-past-the-post divisions (and just eliminated the 111 compensation seats), or if it re-organized the compensation seats at a different level than the state, it should not require an overhaul of OCDIDs, because the first-past-the-post divisions will have retained the same semantics.

sguenther85 commented:
thanks for the information. i will have a look at the comparison between canada and germany.
But I have one more question about the ocd-paths: Why do we have in germany cd's and not ed's?
What is the exact difference?

jpmckinney commented:
I think `cd` (like `state`) was from a time when US types were reused (cd = "congressional district") instead of local names. See https://github.com/opencivicdata/ocd-division-ids/issues/170

`ed` ("electoral district") is the country-neutral term preferred by [Wikipedia](https://en.wikipedia.org/wiki/Electoral_district) and this project for national/subnational divisions. That said, each country can use its own types if appropriate.

For local-level divisions, there is less standardization, e.g. ward, district, borough, division.

I notice that Canada uses `cd` for Census division. In general, users should not assume that types have the same semantics across countries.

azuser commented:
Thanks all for providing context! 

The delimitation of constituencies is proposed by an independent Electoral Districts Commission and validated then by the German Bundestag ([ref](https://www.bundeswahlleiter.de/en/service/glossar/w/wahlkreiskommission.html)). So, in that case, if I rely on previous UK discussions, it makes sense to update the existing hierarchies as proposed previously . Example:

`ocd-division/country:de/state:bb/cd:56` -> `ocd-division/country:de/cd:56`

However, as we decide to update the existing OCD ids, I am wondering if that correction could affect other users.

WDYT ?

sguenther85 commented:
@azuser Your comment last time about the seats which you can get over the state-list was correct.
"So, I guess, the states level is important to represent the electoral system in that case."

So we still need the state level for the candidates who run on these lists or both (state-list and const. list)

azuser commented:
Yeah, I was thinking to keep the states and update the constituencies. So, we will have both states and constituencies attached to `ocd-division/country:de`.

WDYT ?

azuser commented:
I updated the wahlkreise file, please have a look. 

I also updated the aliases file with the new ocd-ids, however, I am not sure if I should do the same for aliases ids (example `ocd-division/country:de/land:bw/wahlkreis:göppingen`)

@jloutsenhizer @jdmgoogle @sguenther85 @jpmckinney WDYT ?

jloutsenhizer commented:
I think the alias IDs shouldn't be changed, clients of the data should use the canonical forms of the OCD IDs.

Only request I'd have for the current state is for aliases to be added for all of the old OCD IDs to the new ones that hang directly off of `ocd-division/country:de`.

azuser commented:
Done. However, I am wondering if at some point we decide to add states constituencies. It can be confusing if we use the same identifiers.

jloutsenhizer commented:
When state constituencies are added, they should use distinct OCD IDs from these previous ones regardless of whether or not these aliases existed. Otherwise there could be data laying around using these ones for the Bundestag constituencies which would then be confused for the state constituencies.

With the additional aliases added, these changes LGTM

azuser commented:
LGTM. Thanks @jloutsenhizer. 

 @jdmgoogle @sguenther85 @jpmckinney other thoughts ?

sguenther85 commented:
No. Looks goot to me.
Now it would be able to append constituencies for the 2021 Saxony-Anhalt state election under the state "Saxony-Anhalt".
Thanks for the Work ;)

azuser commented:
Thanks all, 

@jpmckinney can you please approve my CL and submit it ?

jpmckinney reviewed (commented):
Why are we using `cd` (congressional district) instead of `ed` (electoral district)? Since we're creating new IDs, we might as well use the more sensical type.

azuser commented:
Thanks for checking! I thought it is better to keep the already used type! Happy to change it !

sguenther85 commented:
I would also prefer ed: instead of cd:
Do we have any other blockers here after changing this?
Would be great if i/we can use this new changes in a timely manner 😄

azuser commented:
Sorry for the delay ! @jpmckinney if the change looks good to you, can you please merge it ?
